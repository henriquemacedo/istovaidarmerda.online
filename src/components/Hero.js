import React from "react";
import styled from "styled-components";

import ButtonLink from "./ButtonLink";
import MouseImage from "./MouseImage";

const Wrapper = styled.section`
	position: relative;
	background-color: var(--yellow);
	width: 100vw;
	min-height: 100vh;
	display: flex;

	> div {
		margin: 0 auto;
		display: flex;
		align-items: center;
	}

	.mouse {
		position: absolute;
		bottom: 15px;
		margin-left: 50%;
		transform: translateX(-50%);
	}
`;

const Logo = styled.div`
	margin-right: 50px;
	height: 100%;
	display: flex;
	align-items: center;

	img {
		margin-top: -325px;
		width: auto;
		height: 300px;
	}
`;

const Intro = styled.div`
	h1 {
		font-size: 4rem;
		text-transform: uppercase;
		color: var(--lightBrown);
	}

	p {
		max-width: 600px;
		line-height: var(--lineHeight);
	}

	svg {
		display: block;
		margin-bottom: -15px;
		width: 500px;
		height: auto;

		text {
			font-size: 4rem;
			font-weight: var(--fontBlack);
			text-transform: uppercase;
			line-height: 4rem;
			stroke: var(--default);
			stroke-width: 2px;
			fill: transparent;
		}
	}
`;

const CallToAction = styled.div`
	margin-top: 50px;
	display: flex;

	a {
		min-width: 200px;

		&:not(:last-child) {
			margin-right: 10px;
		}

		&:hover {
			background-color: var(--darkBrown);
		}
	}
`;

const Hero = () => (
	<Wrapper>
		<div>
			<Logo>
				<img src={"/images/ivdm-logo.svg"} alt={"Isto Vai Dar Merda logo"} />
			</Logo>
			<Intro>
				<h1>
					<svg xmlns="http://www.w3.org/2000/svg" width={498} height={85}>
						<text x={0} y={65}>
							Isto Vai Dar
						</text>
					</svg>
					Merda
				</h1>
				<p>
					O <i>Isto Vai Dar Merda</i> é um evento que materializa a máxima “quem avisa, amigo é”, pensado para abordar
					temas que estão no “tipping point” de dar merda mas que ainda podem ser salvos de tão triste fado.
				</p>
				<CallToAction>
					<ButtonLink url="#schedule" value="Agenda" />
					<ButtonLink url="#hein" value="Hein?!" />
				</CallToAction>
			</Intro>
		</div>

		<MouseImage className="mouse" />
	</Wrapper>
);

export default Hero;
