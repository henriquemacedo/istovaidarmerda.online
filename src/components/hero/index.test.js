import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import MatchMediaMock from "jest-matchmedia-mock";
import Hero, { MOUSE_BREAKPOINT, WIDTH_BREAKPOINT } from "./index";
import MockIntersectionObserver from "../../../utils/mocks/mockIntersectionObserver";

/**
 * Builds the Media Query Stub
 *
 * @param {string} widthQuery
 * @param {MatchMediaMock} matchMedia
 */
function setupMediaQuery(widthQuery = WIDTH_BREAKPOINT, matchMedia) {
  const mediaQuery = widthQuery;
  const firstListener = jest.fn();
  const secondListener = jest.fn();
  const mediaQueryList = window.matchMedia(mediaQuery);

  mediaQueryList.addListener((ev) => ev.matches && firstListener());
  mediaQueryList.addListener((ev) => ev.matches && secondListener());

  matchMedia.useMediaQuery(mediaQuery);
}

describe("src/components/hero", () => {
  /**
   * @type {MatchMediaMock}
   */
  let matchMedia;

  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  beforeEach(() => {
    window.IntersectionObserver = MockIntersectionObserver;
  });

  afterEach(() => {
    matchMedia.clear();
  });
  describe("render", () => {
    it("should render without errors", () => {
      setupMediaQuery(WIDTH_BREAKPOINT, matchMedia);
      const component = render(<Hero />);

      expect(component).toMatchSnapshot();
    });

    it("should render a mouse icon", () => {
      setupMediaQuery(MOUSE_BREAKPOINT, matchMedia);

      const component = render(<Hero />);

      expect(component).toMatchSnapshot();
    });
  });
});
