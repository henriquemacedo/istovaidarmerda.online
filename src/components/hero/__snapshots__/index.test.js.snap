// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/hero render should render a mouse icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="styles__Wrapper-sc-1b4b6pk-0 ieLlth hero is-visible"
        data-testid="hero"
      >
        <div
          class="hero__container"
        >
          <figure
            class="styles__Logo-sc-1b4b6pk-1 ewygJF"
          >
            <img
              alt="Isto Vai Dar Merda logo"
              height="auto"
              src="/images/ivdm-logo.svg"
              width="100%"
            />
          </figure>
          <div
            class="styles__Intro-sc-1b4b6pk-2 cSBBCP"
          >
            <h1
              aria-label="Isto Vai Dar Merda"
              class="hero__title"
              data-testid="hero-title"
            >
              Isto Vai Dar
              <br />
               Merda
            </h1>
            <p
              class="hero__description"
              data-testid="hero-description"
            >
              O 
              <i>
                Isto Vai Dar Merda
              </i>
               é um evento que materializa a máxima “quem avisa, amigo é”, pensado para abordar temas que estão no “tipping point” de dar merda mas que ainda podem ser salvos de tão triste fado.
            </p>
            <div
              class="styles__CallToAction-sc-1b4b6pk-3 cA-dOeZ hero__call-to-action"
            >
              <a
                aria-label="Consulta a Agenda"
                class="styles__Button-e7r930-0 fthixk"
                data-testid="button-link"
                href="/#schedule"
                target="_blank"
              >
                Agenda
              </a>
              <a
                aria-label="Sabe mais sobre o evento"
                class="styles__Button-e7r930-0 fthixk"
                data-testid="button-link"
                href="/#hein"
                target="_blank"
              >
                Hein?!
              </a>
            </div>
          </div>
        </div>
        <svg
          class="mouse"
          data-testid="mouse-image"
          height="40"
          viewBox="0 0 30 40"
          width="30"
        >
          <g
            data-name="mouse"
            fill="#212121"
            id="mouse"
            transform="translate(-945 -1010)"
          >
            <path
              d="M17.5 0h-5A12.5 12.5 0 000 12.5v15A12.5 12.5 0 0012.5 40h5A12.5 12.5 0 0030 27.5v-15A12.5 12.5 0 0017.5 0zm8.75 27.5a8.759 8.759 0 01-8.75 8.75h-5a8.759 8.759 0 01-8.75-8.75v-15a8.759 8.759 0 018.75-8.75h5a8.759 8.759 0 018.75 8.75z"
              id="mouse-alt"
              opacity="0.15"
              transform="translate(945 1010)"
            />
            <path
              d="M15 7.5a2.5 2.5 0 00-2.5 2.5v2.5a2.5 2.5 0 005 0V10A2.5 2.5 0 0015 7.5z"
              data-name="wheel"
              id="wheel"
              opacity="0.15"
              transform="translate(945 1010)"
            >
              <animatemotion
                dur="2s"
                path="M 0 8 V 0 Z"
                repeatCount="indefinite"
              />
            </path>
          </g>
        </svg>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="styles__Wrapper-sc-1b4b6pk-0 ieLlth hero is-visible"
      data-testid="hero"
    >
      <div
        class="hero__container"
      >
        <figure
          class="styles__Logo-sc-1b4b6pk-1 ewygJF"
        >
          <img
            alt="Isto Vai Dar Merda logo"
            height="auto"
            src="/images/ivdm-logo.svg"
            width="100%"
          />
        </figure>
        <div
          class="styles__Intro-sc-1b4b6pk-2 cSBBCP"
        >
          <h1
            aria-label="Isto Vai Dar Merda"
            class="hero__title"
            data-testid="hero-title"
          >
            Isto Vai Dar
            <br />
             Merda
          </h1>
          <p
            class="hero__description"
            data-testid="hero-description"
          >
            O 
            <i>
              Isto Vai Dar Merda
            </i>
             é um evento que materializa a máxima “quem avisa, amigo é”, pensado para abordar temas que estão no “tipping point” de dar merda mas que ainda podem ser salvos de tão triste fado.
          </p>
          <div
            class="styles__CallToAction-sc-1b4b6pk-3 cA-dOeZ hero__call-to-action"
          >
            <a
              aria-label="Consulta a Agenda"
              class="styles__Button-e7r930-0 fthixk"
              data-testid="button-link"
              href="/#schedule"
              target="_blank"
            >
              Agenda
            </a>
            <a
              aria-label="Sabe mais sobre o evento"
              class="styles__Button-e7r930-0 fthixk"
              data-testid="button-link"
              href="/#hein"
              target="_blank"
            >
              Hein?!
            </a>
          </div>
        </div>
      </div>
      <svg
        class="mouse"
        data-testid="mouse-image"
        height="40"
        viewBox="0 0 30 40"
        width="30"
      >
        <g
          data-name="mouse"
          fill="#212121"
          id="mouse"
          transform="translate(-945 -1010)"
        >
          <path
            d="M17.5 0h-5A12.5 12.5 0 000 12.5v15A12.5 12.5 0 0012.5 40h5A12.5 12.5 0 0030 27.5v-15A12.5 12.5 0 0017.5 0zm8.75 27.5a8.759 8.759 0 01-8.75 8.75h-5a8.759 8.759 0 01-8.75-8.75v-15a8.759 8.759 0 018.75-8.75h5a8.759 8.759 0 018.75 8.75z"
            id="mouse-alt"
            opacity="0.15"
            transform="translate(945 1010)"
          />
          <path
            d="M15 7.5a2.5 2.5 0 00-2.5 2.5v2.5a2.5 2.5 0 005 0V10A2.5 2.5 0 0015 7.5z"
            data-name="wheel"
            id="wheel"
            opacity="0.15"
            transform="translate(945 1010)"
          >
            <animatemotion
              dur="2s"
              path="M 0 8 V 0 Z"
              repeatCount="indefinite"
            />
          </path>
        </g>
      </svg>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`src/components/hero render should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="styles__Wrapper-sc-1b4b6pk-0 ieLlth hero is-visible"
        data-testid="hero"
      >
        <div
          class="hero__container"
        >
          <figure
            class="styles__Logo-sc-1b4b6pk-1 ewygJF"
          >
            <img
              alt="Isto Vai Dar Merda logo"
              height="auto"
              src="/images/ivdm-logo.svg"
              width="100%"
            />
          </figure>
          <div
            class="styles__Intro-sc-1b4b6pk-2 cSBBCP"
          >
            <h1
              aria-label="Isto Vai Dar Merda"
              class="hero__title"
              data-testid="hero-title"
            >
              <svg
                height="85"
                width="498"
                xmlns="http://www.w3.org/2000/svg"
              >
                <text
                  x="0"
                  y="65"
                >
                  Isto Vai Dar
                </text>
              </svg>
              <span
                class="hero__title__markup"
              >
                Merda
              </span>
            </h1>
            <p
              class="hero__description"
              data-testid="hero-description"
            >
              O 
              <i>
                Isto Vai Dar Merda
              </i>
               é um evento que materializa a máxima “quem avisa, amigo é”, pensado para abordar temas que estão no “tipping point” de dar merda mas que ainda podem ser salvos de tão triste fado.
            </p>
            <div
              class="styles__CallToAction-sc-1b4b6pk-3 cA-dOeZ hero__call-to-action"
            >
              <a
                aria-label="Consulta a Agenda"
                class="styles__Button-e7r930-0 fthixk"
                data-testid="button-link"
                href="/#schedule"
                target="_blank"
              >
                Agenda
              </a>
              <a
                aria-label="Sabe mais sobre o evento"
                class="styles__Button-e7r930-0 fthixk"
                data-testid="button-link"
                href="/#hein"
                target="_blank"
              >
                Hein?!
              </a>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="styles__Wrapper-sc-1b4b6pk-0 ieLlth hero is-visible"
      data-testid="hero"
    >
      <div
        class="hero__container"
      >
        <figure
          class="styles__Logo-sc-1b4b6pk-1 ewygJF"
        >
          <img
            alt="Isto Vai Dar Merda logo"
            height="auto"
            src="/images/ivdm-logo.svg"
            width="100%"
          />
        </figure>
        <div
          class="styles__Intro-sc-1b4b6pk-2 cSBBCP"
        >
          <h1
            aria-label="Isto Vai Dar Merda"
            class="hero__title"
            data-testid="hero-title"
          >
            <svg
              height="85"
              width="498"
              xmlns="http://www.w3.org/2000/svg"
            >
              <text
                x="0"
                y="65"
              >
                Isto Vai Dar
              </text>
            </svg>
            <span
              class="hero__title__markup"
            >
              Merda
            </span>
          </h1>
          <p
            class="hero__description"
            data-testid="hero-description"
          >
            O 
            <i>
              Isto Vai Dar Merda
            </i>
             é um evento que materializa a máxima “quem avisa, amigo é”, pensado para abordar temas que estão no “tipping point” de dar merda mas que ainda podem ser salvos de tão triste fado.
          </p>
          <div
            class="styles__CallToAction-sc-1b4b6pk-3 cA-dOeZ hero__call-to-action"
          >
            <a
              aria-label="Consulta a Agenda"
              class="styles__Button-e7r930-0 fthixk"
              data-testid="button-link"
              href="/#schedule"
              target="_blank"
            >
              Agenda
            </a>
            <a
              aria-label="Sabe mais sobre o evento"
              class="styles__Button-e7r930-0 fthixk"
              data-testid="button-link"
              href="/#hein"
              target="_blank"
            >
              Hein?!
            </a>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
